File: problems/869/solution.go
package main
func reorderedPowerOf2(N int) bool {
    counter := getCounter(N)
    
    for i := 0; i < 32; i++ {
        if (1 << i == N || getCounter(1 << i) == counter) {
            return true
        }
    }
    
    return false
}

func getCounter(num int) int {
    ans := 0
    
    for ; num > 0; num /= 10 {
        ans += int(math.Pow(10.0, float64(num % 10)))
    }
    
    return ans
}

File: problems/869/solution.cpp
class Solution {
public:
    bool reorderedPowerOf2(int N) {
        vector<int> num = getNum(N);
        
        for (int i = 0; i < 31; i++) {
            int powOf2 = pow(2, i);
            
            if (N == powOf2 or num == getNum(powOf2)) {
                return true;
            }
        }
        
        return false;
    }
    
    vector<int> getNum(int n) {
        vector<int> ans(10);
        
        for (char ch: to_string(n)) {
            ans[ch - '0']++;
        }
        
        return ans;
    }
};